src/core/gui/PageView.cpp

```diff
@@ -382,15 +382,23 @@ auto XojPageView::onButtonPressEvent(const PositionInputData& pos) -> bool {
                h->getToolType() == TOOL_PLAY_OBJECT || h->getToolType() == TOOL_SELECT_OBJECT ||
                h->getToolType() == TOOL_SELECT_PDF_TEXT_LINEAR || h->getToolType() == TOOL_SELECT_PDF_TEXT_RECT) {
         if (h->getToolType() == TOOL_SELECT_RECT) {
-            assert(!selection);
-            this->selection = std::make_unique<RectSelection>(x, y);
-            this->overlayViews.emplace_back(std::make_unique<xoj::view::SelectionView>(
-                    this->selection.get(), this, this->settings->getSelectionColor()));
+            if (!selection) {
+                this->selection = std::make_unique<RectSelection>(x, y);
+                this->overlayViews.emplace_back(std::make_unique<xoj::view::SelectionView>(
+                        this->selection.get(), this, this->settings->getSelectionColor()));
+            } else {
+                assert(settings->getInputSystemTPCButtonEnabled());
+                // the selection has already been created by a stylus button press while the stylus was hovering
```

Comment:
nit: the comment should be folded into the assertion message to make debugging easier.
```suggestion
                assert(settings->getInputSystemTPCButtonEnabled() &&
                       "the selection has already been created by a stylus button press while the stylus was hovering!");
```

Reply:
Thanks, I applied your suggestion.